lib/api/CRetrainableModelJsonReader.cc

```diff
@@ -22,15 +21,219 @@
 #include <api/CDataSummarizationJsonWriter.h>
 #include <api/CInferenceModelDefinition.h>
 
-#include <rapidjson/document.h>
-#include <rapidjson/rapidjson.h>
+#include <boost/json.hpp>
+// This file must be manually included when
+// using basic_parser to implement a parser.
+#include <boost/json/basic_parser_impl.hpp>
 
 #include <memory>
 #include <sstream>
 #include <stdexcept>
 #include <string>
 #include <utility>
 
+namespace json = boost::json;
+
+namespace {
+class custom_parser {
+    struct handler {
+        static inline std::string IDENTITY_ENCODING_TAG = "identity_encoding";
+        static inline std::string ONE_HOT_ENCODING_TAG = "one_hot_encoding";
+        static inline std::string FREQUENCY_ENCODING_TAG = "frequency_encoding";
+        static inline std::string TARGET_MEAN_ENCODING_TAG = "target_mean_encoding";
+
+        constexpr static std::size_t max_object_size = std::size_t(-1);
+        constexpr static std::size_t max_array_size = std::size_t(-1);
+        constexpr static std::size_t max_key_size = std::size_t(-1);
+        constexpr static std::size_t max_string_size = std::size_t(-1);
```

Comment:
`std::numeric_limits<std::size_t>::max()` would be less hacky.

Alternatively, maybe we _should_ be setting some limits that are lower than the absolute max, yet still very large. For example, I don't think we'd ever expect a key longer than 1KB, or a string longer than 1GB. And I cannot imagine an object or array with more than a million elements. Any larger values would likely indicate corrupt input.

Reply:
I'll switch to using `std::numeric_limits<std::size_t>::max()` for now and experiment with lower values in a follow up PR.